scipy/signal/filter_design.py

```diff
@@ -3504,6 +3504,196 @@ def besselap(N, norm='phase'):
     return asarray([]), asarray(p, dtype=complex), float(k)
 
 
+def iirnotch(w0, bw, ab=3):
+    """
+    Design second-order IIR notch digital filter.
+
+    Parameters
+    ----------
+    w0 : float
+        A scalar that must satisfy  0 < w0 < 1. For notch filters,
+        w0 is the normalized frequency to remove from the filtered
+        signal (w0 = 1 corresponds to half of the sample frequency).
+    bw : float
+        Filter bandwidth.
+    ab : float, optional
+        Bandwidht maximum atenuation in decibels. By default, `ab`
+        is 3, so the atenuation is -3 dB.
+
+    Returns
+    -------
+    b, a : ndarray, ndarray
+        Numerator (`b`) and denominator (`a`) polynomials
+        of the IIR filter.
+
+    Notes
+    -----
+    The quality factor (Q factor) for the filter is related to the
+    filter bandwidth by Q = w0/bw (for the default value of `ab`).
+
+    .. versionadded: 0.19.0
+
+    References
+    ----------
+    .. [1] Sophocles J. Orfanidis, "Introduction To Signal Processing",
+           Prentice-Hall, 1996
+
+    Examples
+    --------
+    Design and plot filter to remove the 60Hz component from a
+    signal sampled at 1000Hz, using a quality factor Q=30
+
+    >>> from scipy import signal
+    >>> import matplotlib.pyplot as plt
+
+    >>> fs = 1000.0  # Sample frequency (Hz)
+    >>> f0 = 60.0  # Frequency to be removed from signal (Hz)
+    >>> q = 30.0  # Quality factor
+    >>> w0 = f0/(fs/2) # Normalized Frequency
+    >>> bw = w0/q # Bandwidth
+    >>> # Design notch filter
+    >>> b, a = signal.iirnotch(w0, bw)
+
+    >>> # Frequency response
+    >>> w, h = signal.freqz(b, a, 2**14)
+    >>> # Generate frequency axis
+    >>> freq = w*fs/(2*np.pi)
+    >>> # Plot
+    >>> fig, ax = plt.subplots()
+    >>> ax.plot(freq, 20*np.log10(abs(h)), color='black', linewidth=3)
+    >>> ax.grid()
+    >>> ax.set_title("Frequency Response", fontsize=20)
+    >>> ax.set_ylabel("Amplitude (dB)", fontsize=18)
+    >>> ax.set_xlabel("Frequency (Hz)", fontsize=18)
+    >>> ax.axis([0, 100, -50, 10])
+    >>> plt.show()
+    """
+
+    # Guarantee that the inputs are floats
+    w0 = float(w0)
+    bw = float(bw)
+    ab = float(ab)
+
+    # Checks if w0 is within the range
+    if w0 > 1.0 or w0 < 0.0:
+        raise ValueError("w0 should be such that 0 < w0 < 1")
+
+    # Normalize inputs
+    bw = bw*np.pi
+    w0 = w0*np.pi
+
+    # Compute atenuation from ab
+    gb = 10.0**(-ab/20.0)
+
+    # Compute beta: formula 11.3.4 from reference [1]
+    beta = (np.sqrt(1.0-gb**2.0)/gb)*np.tan(bw/2.0)
+
+    # Compute gain: formula 11.3.6 from reference [1]
+    gain = 1.0/(1.0+beta)
+
+    # Compute numerator b and denominator a
+    # formula 11.3.7 from reference [1]
+    b = gain*np.array([1.0, -2.0*np.cos(w0), 1.0])
+    a = np.array([1.0, -2.0*gain*np.cos(w0), (2.0*gain-1.0)])
```

Comment:
So this method always produces a second-order filter, then? Please add a note to the docs.